### ISSUE TITLE
Issue with redrawing chart's legend when external zoom is applied

### ISSUE THREAD

**Issue by Lackard** [amcharts/amcharts5#927]
I'm developing charts that can be used in an external web application that creates some dashboards. Most of the dashboards consists of multiple charts so in order to show them all at once the web application has it's own buttons for zooming the whole view. Unfortunately the zooming is achieved by applying a transform matrix to the div element that contains all of the charts. This obviously introduced the problem with cursor's offset but I've dealt with that by adding the recommended resize function to the root element:
`root = am5.Root.new(rootContainer, { calculateSize: function(dimensions) {
    return {
				  width: rootContainer.clientWidth,
				  height: rootContainer.clientHeight
				};
}});`
This solved the problem with the offset however I'm still facing an issue with proper resizing of chart's legend and labels.
The original chart looks like this:
![image](https://github.com/amcharts/amcharts5/assets/44086952/45406c1e-4d52-4089-86d7-916c0558f7d5)

After applying the external zoom (div's style: `transform: matrix(0.49, 0, 0, 0.49, 0, 0)`) the chart looks like this:
![image](https://github.com/amcharts/amcharts5/assets/44086952/d29ea804-b884-4b5c-9712-29526a564126)

As seen above the tooltips and labels are not scaled and the legend takes up disproportionately more space than the graph (mainly because the layout changed automatically from grid to column). Also for me it seems that the labels in the legend should be smaller. 

What's interesting is that the older chart written with amcharts 4 (right) is scaled properly:
![image](https://github.com/amcharts/amcharts5/assets/44086952/63086834-7ef8-4ec8-b9b9-848e97acd827)

I tried to move the legend to a different container and create a legendRoot element with calculateSize function however it didn't really work. Is there any way to make legend and other elements scale proportionally with graph without changing the layout of the chart?

**Comment by martynasma** [comment]
Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by Lackard** [comment]
I've managed to roughly recreate the conditions. Here is the link:
https://jsfiddle.net/Lackard/67vcdsu0/50/

**Comment by martynasma** [comment]
Thanks. Though I don't see any issues with the legend when zooming in/out:
<img width="650" alt="chrome_2023-05-25_18-52-58" src="https://github.com/amcharts/amcharts5/assets/1458520/18d2483d-1912-4047-8e8e-6ecc05812025">
<img width="539" alt="chrome_2023-05-25_18-51-26" src="https://github.com/amcharts/amcharts5/assets/1458520/53ea5938-75d1-4a0b-804d-a7ad911197a9">
<img width="621" alt="chrome_2023-05-25_18-52-30" src="https://github.com/amcharts/amcharts5/assets/1458520/443f53be-eee9-4fce-89b2-a61f059ff484">
<img width="638" alt="chrome_2023-05-25_18-52-43" src="https://github.com/amcharts/amcharts5/assets/1458520/35fe6d86-1018-42a5-b508-16160824658e">

**Comment by Lackard** [comment]
Please zoom it out a little bit more :) :
![image](https://github.com/amcharts/amcharts5/assets/44086952/ccb6e8e5-1afb-4ed3-948a-44c653c98ca0)
![image](https://github.com/amcharts/amcharts5/assets/44086952/e74f216f-0f96-42d0-b6a6-6c984948a334)

In those cases the graph becomes unreadable and the legend doesn't fit in the main container.
Also in the previous examples you can still see that the legend elements don't really change their size. Instead they are getting wrapped up (it looks like their container is getting smaller but at the same time the elements stay the same size).

What I would like to achieve is to properly redraw whole chart in a proper scale (together with tooltips, labels, etc.) without changing it's layout and to avoid cursor's offset at the same time:
![image](https://github.com/amcharts/amcharts5/assets/44086952/45b2b221-b849-4120-aa1f-e9b6835f9ce2)

**Comment by martynasma** [comment]
This is by design. There's really no way to scale an amCharts 5 chart using CSS. You can scale the div, but the chart itself will adjust for the new dimensions, but it will still draw normal-sized elements.

We're currently experimenting with a way to truly scale the chart, but there's no ETA or guarantee it will ever work.

If it does, I'll let you know here.

That said, scaling down the chart as it is, is not a very good UX, since it reduces readability and user's ability to interact with the chart effectively.

**Comment by Lackard** [comment]
Yeah, I agree that this css scaling is really problematic. Unfortunately I can't do anything about it since that's how the external web application works and I need to get in line with that.

Thank you for your feedback and please let me know if anything changes.
